File: problems/problems_LCR_075/Solution.java
package problems.problems_LCR_075;

import com.alibaba.fastjson.JSON;
import java.util.*;
import qubhjava.BaseSolution;


public class Solution extends BaseSolution {
    public int[] relativeSortArray(int[] arr1, int[] arr2) {

    }

    @Override
    public Object solve(String[] inputJsonValues) {
        int[] arr1 = jsonArrayToIntArray(inputJsonValues[0]);
		int[] arr2 = jsonArrayToIntArray(inputJsonValues[1]);
        return JSON.toJSON(relativeSortArray(arr1, arr2));
    }
}

File: problems/problems_LCR_075/solution.go
package problemLCR_075

import (
	"encoding/json"
	"log"
	"strings"
)

func relativeSortArray(arr1 []int, arr2 []int) (ans []int) {
	mx := 0
	for _, v := range arr1 {
		mx = max(mx, v)
	}
	cnt := make([]int, mx+1)
	for _, v := range arr1 {
		cnt[v]++
	}
	for _, v := range arr2 {
		for cnt[v] > 0 {
			ans = append(ans, v)
			cnt[v]--
		}
	}
	for i := 0; i <= mx; i++ {
		for cnt[i] > 0 {
			ans = append(ans, i)
			cnt[i]--
		}
	}
	return
}

func Solve(inputJsonValues string) interface{} {
	inputValues := strings.Split(inputJsonValues, "\n")
	var arr1 []int
	var arr2 []int

	if err := json.Unmarshal([]byte(inputValues[0]), &arr1); err != nil {
		log.Fatal(err)
	}
	if err := json.Unmarshal([]byte(inputValues[1]), &arr2); err != nil {
		log.Fatal(err)
	}

	return relativeSortArray(arr1, arr2)
}
